src/modals/modal-overlay/modal-overlay.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.modals.componenet:pfModalOverlay
+ *
+ * @param {function} ng-click The on click even handler which invokes the Angular UI $uibModal service. the $uibModal
+ * service only has one method open(options).
+ * See the {@link https://angular-ui.github.io/bootstrap/#modal Angular UI modal} documentation for how to configure the modal.
+ * At a minimum the Angular Patternfly implementation requires two options passed to $uibModal's open method:
+ * <ul>
+ *     <li>component: 'pfModalOverlay'</li>
+ *     <li>resolve: {...}</li>
+ * </ul>
+ * The resolve contains all the members that will be resolved and passed to the controller as locals. Each member is
+ * a function that returns the desired values. The pfModalOverlay controller expects the following members to render the modal:
+ * <ul>
+ *   <li>modalId - string  id of the modal</li>
+ *   <li>modalTitle - string modal title displayed in its header</li>
+ *   <li>titleId (optional) - string id of the title of the modal. Customizable so multiple modals can have unique ids.
+ * Set to 'modalTitle' by default</li>
+ *   <li>hideCloseIcon (optional) - boolean hide-close-icon Hides the 'x' close button, default is false</li>
+ *   <li>modalBody - html string to display in the modal</li>
+ *   <li>actionButtons - Array of button objects. Each button object can have the following properties:
+ *      <ul style='list-style-type: none'>
+ *        <li>.label      - the text to display on the button
+ *        <li>.class      - (optional) classes to add to the button
+ *        <li>.disabled   - (optional) boolean true if the button should be disabled by default
```

Comment:
We should make this consistent with params in other components.  Pls see: https://www.patternfly.org/angular-patternfly/#/api/patternfly.toolbars.component:pfToolbar

Params like 'isDisabled', 'actionFn', and 'onClick'